components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -92,6 +93,26 @@ pub fn rtrim(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
     }))
 }
 
+#[rpn_fn]
+#[inline]
+pub fn left(lhs: &Option<Bytes>, rhs: &Option<Int>) -> Result<Option<Bytes>> {
+    Ok(match (lhs, rhs) {
+        (Some(lhs), Some(rhs)) => {
+            if *rhs <= 0 {
+                return Ok(Some(Vec::new()));
+            }
+            let s = str::from_utf8(&*lhs).unwrap();
```

Comment:
There may be illegal characters. Prefer to handle the error instead of let it panic.